### DISCUSSION TITLE
Raven Embedded Instance Issue

### DISCUSSION THREAD

**Discussion by ryanbuening** [ravendb/ravendb#12601] [Q&A] [answered]
I'm trying to use an embedded instance of RavenDB for local development. I'm adding the embedded instance as follows:

```
EmbeddedServer.Instance.StartServer(new ServerOptions()
{
    DataDirectory = "./localRavenDB",
    ServerUrl = "http://127.0.0.1:8080"
});
```
When I run my .NET Core app with debugging in VS, the embedded instance starts up successfully. When I try to run _without debugging_ in VS, I get the following error:

```
InvalidOperationException: Unable to start the RavenDB Server
Error:
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
File name: 'System.Runtime, Version=6.0.0.0
```

Any ideas on what I'm doing wrong?

**Comment by ppekrol** [comment, marked as answer]
Hi @ryanbuening 

The `6.0.0.0` file version is weird. It looks like the process is trying to use the .NET 6 preview DLLs when it should use the .NET 5 runtime (if you are using 5.0+ or .NET Core 3.1 when you are using 4.2). Can you paste the `dotnet --info` for us?

**Reply by ryanbuening** [reply]
```
.NET SDK (reflecting any global.json):
 Version:   6.0.100-preview.6.21355.2
 Commit:    7f8e0d76c0

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-preview.6.21355.2\

Host (useful for support):
  Version: 6.0.0-preview.6.21352.12
  Commit:  770d630b28

.NET SDKs installed:
  3.1.302 [C:\Program Files\dotnet\sdk]
  3.1.411 [C:\Program Files\dotnet\sdk]
  5.0.100-rc.1.20452.10 [C:\Program Files\dotnet\sdk]
  5.0.101 [C:\Program Files\dotnet\sdk]
  5.0.104 [C:\Program Files\dotnet\sdk]
  5.0.203 [C:\Program Files\dotnet\sdk]
  5.0.205 [C:\Program Files\dotnet\sdk]
  5.0.301 [C:\Program Files\dotnet\sdk]
  5.0.302 [C:\Program Files\dotnet\sdk]
  6.0.100-preview.6.21355.2 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-rc.1.20451.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-preview.6.21355.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-rc.1.20451.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-preview.6.21352.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-rc.1.20452.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0-preview.6.21353.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Reply by ppekrol** [reply]
We will try to reproduce this locally and keep you posted. I'm suspecting that maybe the preview SDKs might be problematic.

Same thing happens when you are running your app from CLI instead of VS?

**Reply by ryanbuening** [reply]
> Same thing happens when you are running your app from CLI instead of VS?

Yes.

**Reply by ppekrol** [reply]
I'm having trouble reproduing this. Here is what I did:

1. Created .NET 3.1 Console App
2. Installed RavenDB Embedded 5.2.1
3. Code:

```csharp
        public static void Main(string[] args)
        {
            EmbeddedServer.Instance.StartServer(new ServerOptions()
            {
                DataDirectory = "./localRavenDB",
                ServerUrl = "http://127.0.0.1:8080"
            });

            using (var store = EmbeddedServer.Instance.GetDocumentStore("DB1"))
            {
                Console.WriteLine("OK");
                Console.ReadLine();
            }
        }
```

4. Installed .NET 6.0 SDKs along with VS 2019 Preview
5. Works both from CLI and VS (with and without debugger attached)

Afterwards I've done some matrix testing:
1. Moved to .NET 5 console app and tested in VS 2019 and VS 2019 Preview + CLI
2. Moved to .NET 6 console app and tested in VS 2019 and VS 2019 Preview + CLI

@ryanbuening Could I ask you to maybe prepare a small project where it is reproducible for you? Maybe I'm not catching something that affects the behavior.

**Reply by ryanbuening** [reply]
I should have specified I'm using VS 2022 with .NET 6 Preview 6 because according to this [post](https://devblogs.microsoft.com/dotnet/announcing-net-6-preview-6/):

> Starting with .NET 6 Preview 6, we’ve tested and support Visual Studio 2022 Preview 2. Visual Studio 2022

Does it still work if you use VS 2022?

**Reply by ppekrol** [reply]
I've managed to reproduce this with .NET 6 Console + VS 2022. The .NET 5 Console + VS 2022 works fine. It must be related to VS 2022 and the way the project is executed (looks like it hooks to process startup - this is what embedded is doing - and forces .NET 6.0 there?). Since this is a preview version, I think it would be better to contract MS about that problem.

**Reply by ryanbuening** [reply]
If I create a bug report for VS 2022 on Microsoft's side and link them to this thread do you think that will be sufficient? Any other details I should include in my report to Microsoft?

**Reply by ppekrol** [reply]
I do not think that more is needed. It is pretty straight-forward. I will be monitoring the issue that you create and comment/assist MS if necessary.

**Reply by ryanbuening** [reply]
Microsoft ticket submitted [here](https://developercommunity.visualstudio.com/t/Project-Execution-Issue-with-VS-2022/1496825).

**Reply by ryanbuening** [reply]
So it looks like Microsoft closed the ticket I created 😕. Any ideas on how to proceed?

**Reply by ppekrol** [reply]
Submitted to dotnet team [here](https://github.com/dotnet/runtime/issues/58000)
